tests/set_maintenance/test_set_maintenance.py

```diff
@@ -41,6 +41,16 @@ def pulp_client(self):
         return self.pulp_client_controller.client
 
 
+@pytest.fixture(autouse=True)
+def constant_lock_id(monkeypatch):
+    # Locks are usually assigned a random ID to make them unique. For testing
+    # we want to keep this constant to compare logs easier.
+    monkeypatch.setattr(
+        "pubtools.pulplib._impl.model.repository.repo_lock.random.choices",
```

Comment:
Please don't do it this way because it ties the pubtools-pulp test too closely to specific implementation details of pubtools-pulplib.

Even completely innocent refactors in pubtools-pulplib such as renaming repo_lock.py to something else, or changing the code from `import random` to `from random import choices`, would cause this test to start failing.

I suggest rather the following solution: in `CommandTester._normalize_plaintext`, there is already code for a similar case of having to normalize a randomly-generated tmpdir path into a placeholder.  Extend that so it replaces strings of the form `(Submitting|Deleting) lock with id '([^']+)'` with `\1 lock with id '<lock_id>'`.
Then we don't depend on how exactly the lock ID is generated, we only depend on how the log messages look, which is considered less of an implementation detail given that it's visible to end-users.